ee/tabby-ui/app/files/components/chat-side-bar.tsx

```diff
@@ -90,3 +97,74 @@ function Header() {
     </div>
   )
 }
+
+async function buildContextPrompt(
+  language: string | undefined,
+  code: string,
+  path: string | undefined
+) {
+  if (!language || !path) {
+    return []
+  }
+
+  if (code.length < 128) {
+    return []
+  }
+
+  const segments = path.split('/');
+  const repo = segments[0];
+  path = segments.slice(1).join('/');
+
+  const tokens = code.split(/[^\w]/).filter(x => x)
+
+  // FIXME(jueliang): restrict query with `git_url` of `repo`.
```

Comment:
@liangfung as we have `git_url` from `repositoryList`, please fill it in a query like
`git_url:${gitUrl}`, and append it to `AND ${gitUrlQuery}`